elyra/pipeline/tests/test_component_parser_airflow.py

```diff
@@ -0,0 +1,96 @@
+#
+# Copyright 2018-2021 Elyra Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import json
+import os
+
+import jupyter_core.paths
+
+from elyra.pipeline.component import FilesystemComponentReader, UrlComponentReader
+from elyra.pipeline.component_parser_airflow import AirflowComponentParser
+from elyra.pipeline.component_registry import ComponentRegistry
+from types import SimpleNamespace
+
+COMPONENT_CATALOG_DIRECORY = os.path.join(jupyter_core.paths.ENV_JUPYTER_PATH[0], 'components')
+
+
+def _get_resource_path(filename):
+    root = os.path.realpath(os.path.join(os.getcwd(), os.path.dirname(__file__)))
+    resource_path = os.path.join(root, 'resources', 'components', filename)
+
+    return resource_path
+
+
+def test_component_registry_can_load_components_from_catalog():
+    component_catalog = os.path.join(COMPONENT_CATALOG_DIRECORY, 'airflow_component_catalog.json')
+    component_parser = AirflowComponentParser()
+    component_registry = ComponentRegistry(component_catalog, component_parser)
+
+    components = component_registry.get_all_components()
+    assert len(components) > 0
+
+
+def test_parse_airflow_component_file():
+    entry = {
+        'id': 'elyra_op_test-operator_TestOperator',
+        'name': 'Test Operator',
+        'type': FilesystemComponentReader._type,
+        'location': _get_resource_path('airflow_test_operator.py'),
+        'adjusted_id': ''
+    }
+    component_entry = SimpleNamespace(**entry)
+
+    parser = AirflowComponentParser()
+    component = parser.parse(component_entry)[0]
+    properties = ComponentRegistry.to_canvas_properties(component)
```

Comment:
I would suggest having a fixture of sorts that specifies the component registry, one with caching enabled, the other not.  Then each test that references the fixture will perform two passes and both forms of registries are exercised.

Reply:
Since there really isn't a plan to allow switching between `ComponentRegistry` and `CacheComponentRegistry` and since all tests will complete within the default (and currently not configurable) 60-second TTL, we should update all the test references to use `CachedComponentRegistry`.